src/App.tsx

```diff
@@ -1,14 +1,78 @@
+/* eslint-disable no-console */
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import debounce from 'lodash.debounce';
 
-import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
+import { TodoList } from './components/TodoList';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoSelected, setTodoSelected] = useState<Todo | null>(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [query, setQuery] = useState<string>('');
+  const [appliedQuery, setAppliedQuery] = useState<string>('');
```

Comment:
no need to indicate generics for primitives
```suggestion
  const [query, setQuery] = useState('');
  const [appliedQuery, setAppliedQuery] = useState('');
```

Reply:
+